### DISCUSSION TITLE
how can i monitor the milvus?

### DISCUSSION THREAD

**Discussion by byeongjokim** [milvus-io/milvus#27693] [Q&A and General discussion]
I can find the [docs](https://milvus.io/docs/monitor.md#2-Enable-ServiceMonitor) about monitoring milvus.

Is there other methods for monitoring milvus (e.g., search latency) without deploying servicemonitor resource..?

**Comment by yhmo** [comment]
How the milvus monitoring system works:
- Milvus setup a httpserver(port 9091) inside the proxy node
- Prometheus server fetches the metrics from port 9091, the metrics are defined by Prometheus rule
- Grafana is a webserver, it converts the data of Prometheus server to visualize pages for you so you can view the visualized metrics

Milvus RPC interfaces don't support exporting the metrics.
So, if you don't have Prometheus server and Grafana server, theoretically you can fetch the metrics from port 9091 by http requests. But it is difficult to parse the metrics and visualize them, except you are very familiar with Prometheus.

**Reply by byeongjokim** [reply]
i means, using grafana without servicemonitor.
can i visualize the metrics on grafana only using httpserver(port 9091)?

**Reply by yhmo** [reply]
> i means, using grafana without servicemonitor. can i visualize the metrics on grafana only using httpserver(port 9091)?

I think it is impossible.

**Reply by byeongjokim** [reply]
thanks.
